src/cow-react/modules/limitOrders/utils/getFilledAmounts.ts

```diff
@@ -0,0 +1,78 @@
+import { ParsedOrder } from '@cow/modules/limitOrders/containers/OrdersWidget/hooks/useLimitOrdersList'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { BigNumber } from 'bignumber.js'
+import JSBI from 'jsbi'
+
+// TODO: using .toNumber() we potentially lose accuracy
+// TODO: must be refactored with replacing bignumber.js by @ethersproject/bignumber
```

Comment:
If we do migrations to etherjs v6, we should use native ES6 bignumber

Reply:
maybe for now we can just change the TODO

Reply:
Updated todo